### ISSUE TITLE
markdown with footnotes incorrectly formatted

### ISSUE THREAD

**Issue by nschloe** [executablebooks/mdformat-footnote#6]
### Describe the bug

Consider the following Markdown (GitHub footnotes):
```markdown
[^lorem]

[^lorem]:
    ipsum
```
Rendered on GitHub:

![screenshot](https://user-images.githubusercontent.com/181628/183264531-c1b2e187-67b2-48b1-a9d5-9ec68c5f3c03.png)

When running mdformat on it:

```
pip install mdformat mdformat-gfm mdformat-frontmatter mdformat-footnote
mdformat a.md
```
```
Error: Could not format "/tmp/a.md".

The formatted Markdown renders to different HTML than the input Markdown. This
is likely a bug in mdformat. Please create an issue report here, including the
input Markdown: https://github.com/executablebooks/mdformat/issues
```

### Reproduce the bug

See above.

### List your environment

- ```
  jupyter-book --version
  ```
  ```
  zsh: command not found: jupyter-book
  ```

- Python 3.10.5 (main, Jun  8 2022, 09:26:22) [GCC 11.3.0] on linux

- Ubuntu 22.10

- ```
  mdformat --version
  ```
  ```
  mdformat 0.7.14 (mdformat_tables: 0.4.1, mdformat_footnote: 0.1.1, mdformat_gfm: 0.3.5, mdformat_frontmatter: 0.4.1)
  ```

**Comment by hukkin** [comment]
Thanks for the issue!

This seems to be a bug in the footnote plugin so I've moved the issue to `mdformat-footnote` issue tracker.

**Comment by hukkin** [comment]
If you need quick workaround before a proper bug fix, you can manually edit your Markdown to e.g.
```markdown
[^lorem]

[^lorem]: ipsum
```
and it should format just fine.

**Comment by hukkin** [comment]
I think the problem is that the footnote parser (poorly specified, using [Pandoc definition](https://pandoc.org/MANUAL.html#footnotes)) doesn't allow footnotes where the first paragraph of the description starts on the next line after the identifier. So the footnote in the example is not actually a valid footnote. Mdformat-footnote currently manipulates whitespace so that after formatting it is a valid footnote. Mdformat-footnote shouldn't do this, it should retain the original rendered HTML which is
```html
<p><a href="ipsum">^lorem</a></p>
```
i.e. a link reference.

This is an annoying special casing though where we need to preserve a poorly formatted reference link because proper formatting changes semantics. Best would be to convince upstream to make footnote syntax in line with reference link syntax.

**Comment by nschloe** [comment]
> doesn't allow footnotes where the first paragraph of the description starts on the next line after the identifier.

Really? I only find

```
[^longnote]: Here's one with multiple blocks.

    Subsequent paragraphs are indented to show that they
belong to the previous footnote.

        { some.code }

    The whole paragraph can be indented, or just the first
    line.  In this way, multi-paragraph footnotes work like
    multi-paragraph list items.
```

It doesn't say the first line can't be empty.

**Comment by hukkin** [comment]
Haha, yep that's why I think the spec is bad. It doesn't take any stance on whether the first line can be empty or not, and all examples have a non-empty first line.

The parser implementation, [mdit-py-plugins/footnote](https://github.com/executablebooks/mdit-py-plugins/tree/master/mdit_py_plugins/footnote) (which is a Python port of the Javascript [markdown-it-footnote](https://github.com/markdown-it/markdown-it-footnote)) implements the spec in a way where the first line can not be empty, however. These are the upstream parser that should be convinced to change their behavior, IMO, using arguments such as that footnote syntax should follow similar rules as CommonMark reference links, and perhaps that GitHub's parser also allows an empty first line.

**Comment by nschloe** [comment]
> The parser implementation, [mdit-py-plugins/footnote](https://github.com/executablebooks/mdit-py-plugins/tree/master/mdit_py_plugins/footnote) (which is a Python port of the Javascript [markdown-it-footnote](https://github.com/markdown-it/markdown-it-footnote)) implements the spec in a way where the first line can not be empty, however.

Ah, I see the problem. I'll file a bug there and link it here.

**Comment by nschloe** [comment]
> The parser implementation, [mdit-py-plugins/footnote](https://github.com/executablebooks/mdit-py-plugins/tree/master/mdit_py_plugins/footnote) (which is a Python port of the Javascript [markdown-it-footnote](https://github.com/markdown-it/markdown-it-footnote)) implements the spec in a way where the first line can not be empty, however.

Ah, I see the problem. I'll file a bug there and link it here.

**Comment by hukkin** [comment]
So the fix for this issue is https://github.com/executablebooks/mdit-py-plugins/pull/49

It should be sufficient to merge that PR and make a new `mdit-py-plugins` release. No changes should be needed in this repository.

**Comment by hukkin** [comment]
This should now be fixed with `pip install mdit-py-plugins>=0.3.1`.
